src/components/va-text-input/va-text-input.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  Component,
+  Element,
+  Prop,
+  Host,
+  h,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+
+@Component({
+  tag: 'va-text-input',
+  styleUrl: 'va-text-input.css',
+  shadow: true,
+})
+export class VaTextInput {
+  @Element() el: HTMLElement;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * The label for the text input.
+   */
+  @Prop() label: string | HTMLElement;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string | HTMLElement;
+
+  /**
+   * Set the input to required and render the (Required) text
+   */
+  @Prop() required?: boolean;
+
+  /**
+   * Placeholder text to show in the input field
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * The maximum number of characters allowed in the input.
+   */
+  @Prop() maxlength?: number;
+
+  /**
+   * What to tell the browser to auto-complete the field with.
+   */
+  @Prop() autocomplete?: string;
+
+  /**
+   * Don't emit any component-library-analytics events
+   */
+  @Prop() disableAnalytics?: boolean;
+
+  /*
+   * The value for the input.
+   */
+  @Prop({ mutable: true }) value?: string = '';
+  // TODO: Make the value prop reflective. Currently, it isn't because it screws
+  // up the input behavior. For now, the only bug is that the changed value
+  // isn't reflected in the DOM on the web component.
+  //
+  // $('va-text-input').value will be correct
+  // $('va-text-input').getAttribute('value') will be incorrect
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.value = target.value;
+  };
+
+  private handleBlur = () => {
+    if (this.disableAnalytics) return;
+
+    this.componentLibraryAnalytics.emit({
+      componentName: 'va-text-input',
+      action: 'blur',
+      details: {
+        label: this.label,
+        value: this.value,
+      },
+    });
+  };
+
+  render() {
+    const atts = assembleAttributes(this.el.attributes);
+    if (this.error) {
+      atts['aria-describedby'] = (
+        (atts['aria-describedby'] || '') + ' error-message'
+      ).trim();
+    }
+    return (
+      <Host>
+        <div class={this.error ? 'input-wrapper input-error' : 'input-wrapper'}>
```

Comment:
I think you might be able to get rid of this wrapping div entirely. In the CSS you could use an attribute selector like this:

https://github.com/department-of-veterans-affairs/component-library/blob/d32e14d863547300980b4e3ec982dd79d79a4c63/src/components/va-select/va-select.css#L13

And move the `input-error` styles into that.

You could also replace `input-wrapper` with a `:host` selector in the CSS which might remove the need for this div.

Reply:
Ahhhh, `:host([error])`. I like it! :shipit: